$OpenBSD: patch-ipc_chromium_src_chrome_common_ipc_message_utils_h,v 1.5 2012/05/09 17:16:01 landry Exp $
64 bits only
ipc_message_utils.h: In static member function 'static void IPC::ParamTraits<base::Time>::Write(IPC::Message*, const base::Time&)':
ipc_message_utils.h:310: error: 'Write' is not a member of 'IPC::ParamTraits<long long int>'
--- ipc/chromium/src/chrome/common/ipc_message_utils.h.orig	Wed Jan 11 18:14:43 2012
+++ ipc/chromium/src/chrome/common/ipc_message_utils.h	Sat Jan 14 20:21:44 2012
@@ -193,6 +193,29 @@ struct ParamTraits<unsigned long long> {
     l->append(StringPrintf(L"%ull", p));
   }
 };
+
+template <>
+struct ParamTraits<long long> {
+  typedef long long param_type;
+  static void Write(Message* m, const param_type& p) {
+    m->WriteData(reinterpret_cast<const char*>(&p), sizeof(param_type));
+ }
+  static bool Read(const Message* m, void** iter, param_type* r) {
+    const char *data;
+    int data_size = 0;
+    bool result = m->ReadData(iter, &data, &data_size);
+    if (result && data_size == sizeof(param_type)) {
+      memcpy(r, data, sizeof(param_type));
+    } else {
+      result = false;
+      NOTREACHED();
+    }
+    return result;
+  }
+  static void Log(const param_type& p, std::wstring* l) {
+    l->append(StringPrintf(L"%ull", p));
+  }
+};
 #endif
 
 #if !(defined(OS_MACOSX) || defined(OS_OPENBSD) || defined(OS_WIN) || (defined(OS_LINUX) && defined(ARCH_CPU_64_BITS)) || defined(ARCH_CPU_S390))
